Description: Patch to allow injection of hardening flags
 This patch allows the hardening flags injection into all
 Makefile during build time. It also removes -Os from upstream
 config.mk to allow DEB_BUILD_OPTIONS=noopt
 .
Author: Vasudev Kamath <kamathvasudev@gmail.com>
Last-Update: 2012-07-29

--- suckless-tools-39.1.orig/lsw/config.mk
+++ suckless-tools-39.1/lsw/config.mk
@@ -13,9 +13,9 @@ INCS = -I${X11INC}
 LIBS = -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS   = -ansi -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS  = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS   += -ansi -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS  += -s ${LIBS}
 
 # compiler and linker
 CC = cc
--- suckless-tools-39.1.orig/swarp/config.mk
+++ suckless-tools-39.1/swarp/config.mk
@@ -15,8 +15,8 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CFLAGS = -Os ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS}
+CFLAGS += ${INCS} -DVERSION=\"${VERSION}\"
+LDFLAGS += ${LIBS}
 #CFLAGS = -g -Wall -O2 ${INCS} -DVERSION=\"${VERSION}\"
 #LDFLAGS = -g ${LIBS}
 
--- suckless-tools-39.1.orig/lsx/config.mk
+++ suckless-tools-39.1/lsx/config.mk
@@ -12,8 +12,8 @@ INCS = -I/usr/lib
 LIBS = -L/usr/lib -lc
 
 # flags
-CFLAGS = -Os ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS}
+CFLAGS += ${INCS} -DVERSION=\"${VERSION}\"
+LDFLAGS += ${LIBS}
 #CFLAGS = -g -Wall -O2 ${INCS} -DVERSION=\"${VERSION}\"
 #LDFLAGS = -g ${LIBS}
 
--- suckless-tools-39.1.orig/tabbed/config.mk
+++ suckless-tools-39.1/tabbed/config.mk
@@ -12,9 +12,9 @@ INCS = -I. -I/usr/include
 LIBS = -L/usr/lib -lc -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -s ${LIBS}
 
 # Solaris
 #CFLAGS = -fast ${INCS} -DVERSION=\"${VERSION}\"
--- suckless-tools-39.1.orig/sprop/config.mk
+++ suckless-tools-39.1/sprop/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -s ${LIBS}
 
 # compiler and linker
 CC = cc
--- suckless-tools-39.1.orig/ssid/config.mk
+++ suckless-tools-39.1/ssid/config.mk
@@ -11,8 +11,8 @@ INCS = -I. -I/usr/include
 LIBS = -L/usr/lib -lc
 
 # flags
-CFLAGS = -Os ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS}
+CFLAGS += ${INCS} -DVERSION=\"${VERSION}\"
+LDFLAGS += ${LIBS}
 #CFLAGS = -g -Wall -O2 ${INCS} -DVERSION=\"${VERSION}\"
 #LDFLAGS = -g ${LIBS}
 
--- suckless-tools-39.1.orig/dmenu/config.mk
+++ suckless-tools-39.1/dmenu/config.mk
@@ -17,10 +17,10 @@ INCS = -I${X11INC}
 LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS}
 
 # flags
-CPPFLAGS = -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
+CPPFLAGS += -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS   = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS  = -s ${LIBS}
+CFLAGS   += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS  += -s ${LIBS}
 
 # compiler and linker
 CC = cc
--- suckless-tools-39.1.orig/slock/config.mk
+++ suckless-tools-39.1/slock/config.mk
@@ -14,9 +14,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -lcrypt -L${X11LIB} -lX11 -lXext
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\" -DHAVE_SHADOW_H
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\" -DHAVE_SHADOW_H
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -s ${LIBS}
 
 # On *BSD remove -DHAVE_SHADOW_H from CPPFLAGS and add -DHAVE_BSD_AUTH
 # On OpenBSD and Darwin remove -lcrypt from LIBS
--- suckless-tools-39.1.orig/wmname/config.mk
+++ suckless-tools-39.1/wmname/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -s ${LIBS}
 
 # compiler and linker
 CC = cc
--- suckless-tools-39.1.orig/sselp/config.mk
+++ suckless-tools-39.1/sselp/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -s ${LIBS}
 
 # compiler and linker
 CC = cc
